homeassistant/components/alexa_local_control.py

```diff
@@ -0,0 +1,527 @@
+<minor>0</minor>
+</specVersion>
+<URLBase>http://{0}:{1}/</URLBase>
+<device>
+<deviceType>urn:schemas-upnp-org:device:Basic:1</deviceType>
+<friendlyName>HA-Echo ({0})</friendlyName>
+<manufacturer>Royal Philips Electronics</manufacturer>
+<manufacturerURL>http://www.philips.com</manufacturerURL>
+<modelDescription>Philips hue Personal Wireless Lighting</modelDescription>
+<modelName>Philips hue bridge 2015</modelName>
+<modelNumber>BSB002</modelNumber>
+<modelURL>http://www.meethue.com</modelURL>
+<serialNumber>1234</serialNumber>
+<UDN>uuid:2f402f80-da50-11e1-9b23-001788255acc</UDN>
+<presentationURL>index.html</presentationURL>
+<iconList>
+<icon>
+<mimetype>image/png</mimetype>
+<height>48</height>
+<width>48</width>
+<depth>24</depth>
+<url>hue_logo_0.png</url>
+</icon>
+<icon>
+<mimetype>image/png</mimetype>
+<height>120</height>
+<width>120</width>
+<depth>24</depth>
+<url>hue_logo_3.png</url>
+</icon>
+</iconList>
+</device>
+</root>
+"""
+
+CONF_HOST_IP = 'host_ip'
+CONF_LISTEN_PORT = 'listen_port'
+CONF_OFF_MAPS_TO_ON_DOMAINS = 'off_maps_to_on_domains'
+CONF_EXPOSE_BY_DEFAULT = 'expose_by_default'
+CONF_EXPOSED_DOMAINS = 'exposed_domains'
+
+ATTR_ECHO = 'echo'
+ATTR_ECHO_FRIENDLY_NAME = 'echo_friendly_name'
+
+DEFAULT_LISTEN_PORT = 8300
+DEFAULT_OFF_MAPS_TO_ON_DOMAINS = ['script', 'scene']
+DEFAULT_EXPOSE_BY_DEFAULT = True
+DEFAULT_EXPOSED_DOMAINS = [
+    'switch', 'light', 'script', 'scene', 'group', 'input_boolean',
+    'media_player'
+]
+
+
+def setup(hass, config):
+    """Activate the alexa_local_control component."""
+    from flask import Flask
+
+    app = Flask(__name__)
+    app.url_map.strict_slashes = False
+
+    view = HueBridgeView(hass, config)
+    view.set_up_routes(app)
+
+    def start_listener(event):
+        """Start listening for UPNP/SSDP requests."""
+        upnp_listener = UPNPResponderThread(
+            view.host_ip_addr, view.listen_port)
+        upnp_listener.start()
+
+    def start_web_server():
+        """Start the Hue bridge API web server."""
+        from cherrypy import wsgiserver
+
+        dispatcher = wsgiserver.WSGIPathInfoDispatcher({'/': app})
+        server = wsgiserver.CherryPyWSGIServer(
+            ('0.0.0.0', view.listen_port), dispatcher)
+
+        server.start()
+
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, start_listener)
+    threading.Thread(target=start_web_server, daemon=True, name=DOMAIN).start()
+
+    return True
+
+
+class HueBridgeView(object):
+    """An instance of an emulated Hue bridge."""
+
+    def __init__(self, hass_instance, config):
+        """Initialize the instance."""
+        self.hass = hass_instance
+        self.cached_states = {}
+
+        self.fill_with_config(config)
+
+    def fill_with_config(self, config):
+        """Fill the instance with data from the ocnfiguration file."""
+        # Get the IP address that will be passed to the Echo during discovery
+        self.host_ip_addr = config[DOMAIN].get(CONF_HOST_IP)

```

Comment:
Instead of requiring the user to hardcode the IP address, you can use the [`util.get_local_ip`](https://github.com/robbiet480/home-assistant/blob/dev/homeassistant/util/__init__.py#L84-L97) function.